### DISCUSSION TITLE
Is it possible add  `Commnad()` in `Header` function?

### DISCUSSION THREAD

**Discussion by Jaehaks** [sxyazi/yazi#1618] [1. Q&A] [answered]
Hi

I try to execute git command with `Command()` in Header children
I define this function in setup() 
```lua
function Header:gitbranch()
  local status, err = Command('git')
                      :args({'--no-optional-locks', '-c', 'core.quotePath='})
                      :args({'status'})
                      :args({'--ignore-submodules=dirty', '--branch', '--show-stash', '--ahead-behind'})
                      :args({'2>nul'})
                      :stdout(Command.PIPED)
                      :output()

  local branch = get_branch(status)  -- get branch string 
  return ui.Line{
    ui.Span(branch)
  }
end
```

It doesn't work without any response. 
If I set the command with `io.popen` instead of `Commnad()`, it works well.
but `io.popen` is too slow in Windows to call this function periodically.

Is there any way to execute git command asynchronously?

**Comment by AnirudhG07** [comment, marked as answer]
If you want to show git branch name in the header, you can check out [git.yazi](https://gitee.com/DreamMaoMao/git.yazi), see how it is implemented there. I think you should make a seperate function and then do `Header.children_add(...)` for it to work as you want. You can also use [starship.yazi](https://github.com/Rolv-Apneseth/starship.yazi), since it automatically shows git branch in a nice formatted way.
